helix-term/src/commands.rs

```diff
@@ -2889,3 +3027,30 @@ pub fn right_bracket_mode(cx: &mut Context) {
         }
     })
 }
+
+impl fmt::Display for Command {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let Command(_, name) = self;
+        f.write_str(name)
+    }
+}
+
+impl std::str::FromStr for Command {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        for command in Command::COMMAND_LIST {
+            if command.1 == s {
+                return Ok(command.clone());
+            }
+        }
+        Err(anyhow!("No command named '{}'", s))
```

Comment:
```suggestion
        Command::COMMAND_LIST.iter().find(|cmd| cmd.1 == s).or_else(|s| anyhow!("No command named '{}'", s))
```
I think an iterator based approach would be cleaner here?